code/game/objects/structures/watercloset.dm

```diff
@@ -480,6 +459,175 @@
 		return
 	return ..()
 
+//Legacy sink and puddles, use the type legacysink for unlimited water sources like classic sinks.
```

Comment:
Seeing as the application in this case isn't actually sinks anymore, but the idea of an endless source of a certain reagent, maybe it'd be better to call this type water_source, so it doesn't seem as deprecated

Reply:
also please use autodoc, you can find info on it here https://github.com/SpaceManiac/SpacemanDMM/blob/master/src/dmdoc/README.md

Reply:
A good idea! making the change